cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -109,11 +122,20 @@ public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
   private ListenableFuture<?> execute(final Id.Program id, Map<String, String> sysArgs,
                                       Map<String, String> userArgs) throws Exception {
     ProgramRuntimeService.RuntimeInfo runtimeInfo;
+    String originalUserId = SecurityRequestContext.getUserId();
     try {
+      // if the program has a namespace user configured then set that user in the security request context.
+      // See: CDAP-7396
+      String nsPrincipal = namespaceQueryAdmin.get(id.getNamespace()).getConfig().getPrincipal();
+      if (nsPrincipal != null && SecurityUtil.isKerberosEnabled(cConf)) {
+        SecurityRequestContext.setUserId(new KerberosName(nsPrincipal).getServiceName());
```

Comment:
I see. Yes, I agree it will be setting the value more than necessary but since it's just an assignment shouldn't impact any performance. In the above the two try block and an additional if look a lot more than the advantage so I will keep it as it is since you  don't have a problem with this either.